decomposer/neuro_san_solver.py

```diff
@@ -215,13 +215,16 @@ async def _solve_generic(self, problem: str, source: str) -> tuple[str, list[str
         """
         solutions: list[str] = []
         finals: list[str] = []
+        tool_args: dict[str, Any] = {
+            "problem": problem
+        }
         for k in range(self.solution_candidate_count):
-            r: str = await self.problem_solver_caller.call_agent(problem)
+            r: str = await self.problem_solver_caller.call_agent(tool_args)
             solutions.append(r)
             finals.append(self.parsing.extract_final(r))
             logging.info(f"{source} candidate {k + 1}: {finals[-1]}")
 
-        voter: Voter = FirstToKVoter(source, "composition", self.composition_discriminator_caller,
+        voter: Voter = FirstToKVoter(source, "composition", "solutions", self.composition_discriminator_caller,
```

Comment:
Pass in the appropriate key for the candidates given the agent we are using for the discriminator.